app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,46 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show destroy]
+  before_action :authenticate_user!, except: %i[index show]
```

Comment:
- [ ] **Before Action:** restric access to the **`:new`** and **`:create`** methods ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
```ruby
before_action :signed_in_only!, only: [:new, :create]

def signed_in_only!
  # current_user in ApplicationController must be defined as helper method to make it work
  redirect_to  new_user_session_path unless current_user

  ## alternative solution:
  # if signed_in? is defined in ApplicationController (see stretch requirements)
  redirect_to new_user_session_path unless signed_in?
end
```

Reply:
I have added restrictions on the routes file. However, I'm already using "Authenticate user" flag to only allow the index page. people need to be logged in to access the other pages.